### PR TITLE
better localization 

### PR DESCRIPTION
**Pull request by realjoshuau**
this pr:
- moves the speaker distance to SwerveDrive.java
- removes vision-only pose estimation
- and prevents robot code from crashing on beta or when no speaker tags are in view

i also ran spotless

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -54,13 +54,13 @@ public static final class FeatureFlags {
 
     // public static final boolean kSwerveAccelerationLimitingEnabled = true;
 
-    public static final boolean kLocalizationStdDistanceBased = true;
+    public static final boolean kSwerveUseVisionForPoseEst = false;
```
Comment (realjoshuau) [inline src/main/java/frc/robot/Constants.java:57]: moved bc otw it was confusing

File: src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java [hunk 2]
```diff
@@ -281,9 +302,13 @@ public void localizeThreadCaller() {
     // so localize will update the pose estimator in a separate thread
     // and it'll reflect over to the main thread's poseEstimator
     // when the main thread calls getPose()
+
     while (true) { // using while true instead of periodic()ly creating a new thread!
-      localize("limelight-top");
-      Timer.delay(SwerveConstants.kVisionSecondsBetweenLocalize);
+      if (DriverStation.isTeleopEnabled()) {
```
Comment (realjoshuau) [inline src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java:307]: it only updates vision data in teleop mode

Comment (ThatXliner) [reply src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java:307]: why can't we use localization during auto? Wouldn't it help with note consistency?

**Review by ChaoticChaosTheGreat — COMMENTED** [review]
Because literally everyone on design team has something against global localization, we might want to keep localization for interpolating speaker at least based on just single LL.

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -689,18 +667,5 @@ public void periodic(double dt) {
     }
     ;
     XboxStalker.stalk(driver, operator);
-    // System.out.println(Limelight.getBotpose("limelight").length);
-    double[] visionBotPose = Limelight.getBotpose_wpiBlue("limelight");
```
Comment (ChaoticChaosTheGreat) [inline src/main/java/frc/robot/RobotContainer.java:693]: why not keep this, its for gathering data.

Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/RobotContainer.java:693]: oh cuz it makes it crash

File: src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java (hunk 2 above)
Comment (ChaoticChaosTheGreat) [reply src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java:317]: bruh enable vision in auto and just disable start replanning, but this works

**Review by ChaoticChaosTheGreat — APPROVED** [review]
Nice

**Comment by 3LucasZ** [comment]
Looks pretty good. I will do a deeper review tonight.

**Pull request merged**